docs/ABI/Mangling.rst

```diff
@@ -925,13 +1012,19 @@ now codified into the ABI; the index 0 is therefore reserved.
 
   generic-param-pack-marker ::= 'Rv' GENERIC_PARAM-INDEX   // generic parameter pack marker
 
+  INVERTIBLE-KIND ::= 'c'  // Copyable
+  INVERTIBLE-KIND ::= 'e'  // Escapable
+
   GENERIC-PARAM-COUNT ::= 'z'                // zero parameters
   GENERIC-PARAM-COUNT ::= INDEX              // N+1 parameters
 
   requirement ::= protocol 'R' GENERIC-PARAM-INDEX                  // protocol requirement
   requirement ::= protocol assoc-type-name 'Rp' GENERIC-PARAM-INDEX // protocol requirement on associated type
   requirement ::= protocol assoc-type-list 'RP' GENERIC-PARAM-INDEX // protocol requirement on associated type at depth
   requirement ::= protocol substitution 'RQ'                        // protocol requirement with substitution
+#if SWIFT_RUNTIME_VERSION >= 6.0
+  requirement ::= 'Ri' INVERTIBLE-KIND GENERIC-PARAM-INDEX          // inverse requirement
```

Comment:
Yes I believe we'll need them.

Reply:
@slavapestov seems to think we won’t need them

Reply:
Currently the only place in the language where you can state `T: ~Copyable` with `T` a dependent member type is inside of a protocol, eg

```swift
protocol P {
  associatedtype A: ~Copyable
}
```

I was thinking that for symbol mangling, we don't ever need this because the requirement signature is not mangled as part of any symbol. However, now that I think about it I do believe we mangle the requirement signature in the protocol's context descriptor. So perhaps we need a mangling here after all.

Reply:
Yes, we do mangle the requirement signature as part of the protocol's context descriptor.